components/event-receiver/org.wso2.carbon.event.receiver.core/src/main/java/org/wso2/carbon/event/receiver/core/internal/EventReceiver.java

```diff
@@ -239,7 +248,8 @@ protected void sendEvent(Event event) {
             statisticsMonitor.incrementRequest();
         }
         //in distributed mode if events are duplicated in cluster, send event only if the node is receiver coordinator
-        if (!(mode == Mode.Distributed) || !isEventDuplicatedInCluster || EventReceiverServiceValueHolder.getCarbonEventReceiverManagementService().isReceiverCoordinator()) {
+        if ((!(mode == Mode.Distributed) || !isEventDuplicatedInCluster || EventReceiverServiceValueHolder.getCarbonEventReceiverManagementService().isReceiverCoordinator())
+                        && !isManagerNode) {
```

Comment:
@dilini-muthumala 
Dilini, can we optimize the logic of the if condition so that !isManagerNode is checked only if CEP is running in distributed mode? Checking 'isManagerNode' with an && means that the right operand will still be evaluated if left operand is true. But I think the || conditions have been used here so that if a true condition is matched, the expression becomes true immediately. So can we restructure the if as below? 
```java
    if (mode != Mode.Distributed || !isManagerNode && (!isEventDuplicatedInCluster || isReceiverCoordinator))
```
This way, the isManagerNode check won't happen in non-distributed mode, which might be a minor optimization, but useful since this is in the critical path of sending events. WDYT?

Also, if it is possible to pre-calculate some of these boolean variable combinations in distributed mode, we might be able to reduce the evaluations at the time of sending events. Can you see if it is possible to pre-calculate some of these and variables and optimize this if condition? Some variables such as mode, isEventDuplicatedInCluster won't change within the lifetime of the EventReceiver instance AFAIU, so we might be able to pre-calculate a boolean var accordingly and check that instead of checking multiple conditions at the time of sending the event.